diff -Naur rocksdb-5.9.2/db/c.cc rocksdb-5.9.2-patched/db/c.cc
--- rocksdb-5.9.2/db/c.cc	2017-12-08 03:47:51.000000000 +0800
+++ rocksdb-5.9.2-patched/db/c.cc	2018-10-08 11:54:48.000000000 +0800
@@ -2175,6 +2175,20 @@
   opt->rep.use_fsync = use_fsync;
 }
 
+void rocksdb_options_set_enable_pipelined_write(rocksdb_options_t* opt,
+                                                unsigned char v) {
+  opt->rep.enable_pipelined_write = v;
+}
+
+void rocksdb_options_set_max_subcompactions(rocksdb_options_t* opt, uint32_t v) {
+  opt->rep.max_subcompactions = v;
+}
+
+void rocksdb_options_set_memtable_insert_with_hint_fixed_length_prefix_extractor(rocksdb_options_t* opt,
+                                                                                 size_t length) {
+  opt->rep.memtable_insert_with_hint_prefix_extractor.reset(rocksdb::NewFixedPrefixTransform(length));
+}
+
 void rocksdb_options_set_db_log_dir(
     rocksdb_options_t* opt, const char* db_log_dir) {
   opt->rep.db_log_dir = db_log_dir;
@@ -2753,6 +2767,11 @@
   opt->rep.total_order_seek = v;
 }
 
+void rocksdb_readoptions_set_ignore_range_deletions(rocksdb_readoptions_t* opt,
+                                              unsigned char v) {
+  opt->rep.ignore_range_deletions = v;
+}
+
 rocksdb_writeoptions_t* rocksdb_writeoptions_create() {
   return new rocksdb_writeoptions_t;
 }
@@ -2793,6 +2812,11 @@
   opt->rep.target_level = n;
 }
 
+void rocksdb_compactoptions_set_force_bottommost_level_compaction(
+  rocksdb_compactoptions_t* opt) {
+  opt->rep.bottommost_level_compaction = rocksdb::BottommostLevelCompaction::kForce;
+}
+
 rocksdb_flushoptions_t* rocksdb_flushoptions_create() {
   return new rocksdb_flushoptions_t;
 }
diff -Naur rocksdb-5.9.2/include/rocksdb/c.h rocksdb-5.9.2-patched/include/rocksdb/c.h
--- rocksdb-5.9.2/include/rocksdb/c.h	2017-12-08 03:47:51.000000000 +0800
+++ rocksdb-5.9.2-patched/include/rocksdb/c.h	2018-10-08 11:55:07.000000000 +0800
@@ -838,6 +838,11 @@
 extern ROCKSDB_LIBRARY_API void rocksdb_options_set_use_direct_reads(
     rocksdb_options_t*, unsigned char);
 extern ROCKSDB_LIBRARY_API void
+rocksdb_options_set_enable_pipelined_write(rocksdb_options_t*, unsigned char);
+extern ROCKSDB_LIBRARY_API void
+rocksdb_options_set_max_subcompactions(rocksdb_options_t*, uint32_t);
+extern ROCKSDB_LIBRARY_API void rocksdb_options_set_memtable_insert_with_hint_fixed_length_prefix_extractor(rocksdb_options_t* opt,size_t length);
+extern ROCKSDB_LIBRARY_API void
 rocksdb_options_set_use_direct_io_for_flush_and_compaction(rocksdb_options_t*,
                                                            unsigned char);
 extern ROCKSDB_LIBRARY_API void rocksdb_options_set_is_fd_close_on_exec(
@@ -1057,6 +1062,8 @@
     rocksdb_readoptions_t*, unsigned char);
 extern ROCKSDB_LIBRARY_API void rocksdb_readoptions_set_total_order_seek(
     rocksdb_readoptions_t*, unsigned char);
+extern ROCKSDB_LIBRARY_API void rocksdb_readoptions_set_ignore_range_deletions(
+    rocksdb_readoptions_t*, unsigned char);
 
 /* Write options */
 
@@ -1082,6 +1089,8 @@
     rocksdb_compactoptions_t*, unsigned char);
 extern ROCKSDB_LIBRARY_API void rocksdb_compactoptions_set_target_level(
     rocksdb_compactoptions_t*, int);
+extern ROCKSDB_LIBRARY_API void rocksdb_compactoptions_set_force_bottommost_level_compaction(
+    rocksdb_compactoptions_t*);
 
 /* Flush options */
 
